prov/efa/test/efa_unit_test_rdm_peer.c

```diff
@@ -0,0 +1,135 @@
+/* SPDX-License-Identifier: BSD-2-Clause OR GPL-2.0-only */
+/* SPDX-FileCopyrightText: Copyright Amazon.com, Inc. or its affiliates. All rights reserved. */
+
+#include "efa_unit_tests.h"
+
+/**
+ * @brief Test efa_rdm_peer_reorder_msg
+ *
+ * @param resource efa_resource
+ * 
+ */
+void test_efa_rdm_peer_reorder_msg_impl(struct efa_resource *resource,
+					u_int32_t exp_msg_id, u_int32_t msg_id,
+					int expected_ret)
+{
+	struct efa_rdm_ep *efa_rdm_ep;
+	struct efa_rdm_peer *peer;
+	struct efa_ep_addr raw_addr;
+	fi_addr_t addr;
+	struct efa_unit_test_eager_rtm_pkt_attr pkt_attr = {0};
+	size_t raw_addr_len = sizeof(raw_addr);
+	struct efa_rdm_peer_overflow_pke_list_entry *overflow_pke_list_entry;
+	struct efa_rdm_pke *pkt_entry, *overflow_pkt_entry;
+	struct dlist_entry *tmp;
+	u_int32_t overflow_msg_id;
+	int ret;
+
+	efa_rdm_ep = container_of(resource->ep, struct efa_rdm_ep,
+				  base_ep.util_ep.ep_fid);
+
+	/* Create and register a fake peer */
+	ret = fi_getname(&resource->ep->fid, &raw_addr, &raw_addr_len);
+	assert_int_equal(ret, 0);
+	raw_addr.qpn = 1;
+	raw_addr.qkey = 0x1234;
+	ret = fi_av_insert(resource->av, &raw_addr, 1, &addr, 0 /* flags */, NULL /* context */);
+	assert_int_equal(ret, 1);
+	peer = efa_rdm_ep_get_peer(efa_rdm_ep, addr);
+	assert_non_null(peer);
+	ofi_recvwin_buf_alloc(&peer->robuf, efa_env.recvwin_size);
```

Comment:
This looks weird, aren't recvwin and overflow list created as part of peer construction (in fi_av_insert)? Why do we need to create them explicitly in the test?